pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiNodesOfflineClusterIntegrationTest.java

```diff
@@ -138,28 +140,28 @@ public void testServerHardFailure()
 
     // Take a server and shut down its query server to mimic a hard failure
     BaseServerStarter serverStarter = _serverStarters.get(NUM_SERVERS - 1);
-    serverStarter.getServerInstance().shutDown();
-
-    // First query should hit all servers and get connection refused exception
-    testCountStarQuery(NUM_SERVERS, true);
-
-    // Second query should not hit the failed server, and should return the correct result
-    testCountStarQuery(NUM_SERVERS - 1, false);
-
-    // Restart the failed server, and it should be included in the routing again
-    serverStarter.stop();
-    serverStarter = startOneServer(NUM_SERVERS - 1);
-    _serverStarters.set(NUM_SERVERS - 1, serverStarter);
-    TestUtils.waitForCondition(aVoid -> {
-      try {
+    try {
+      serverStarter.getServerInstance().shutDown();
+
+      // First query should hit all servers and get connection refused or reset exception
+      // TODO: This is a flaky test. There is a race condition between shutDown and the query being executed.
```

Comment:
Let's create an issue to fix this test

Reply:
Created: https://github.com/apache/pinot/issues/10823